addons/stock/models/stock_move.py

```diff
@@ -1293,14 +1290,19 @@ def _prepare_move_line_vals(self, quantity=None, reserved_quant=None):
                 vals = dict(vals, product_uom_qty=uom_quantity)
             else:
                 vals = dict(vals, product_uom_qty=quantity, product_uom_id=self.product_id.uom_id.id)
+        package = None
         if reserved_quant:
+            package = reserved_quant.package_id
             vals = dict(
                 vals,
                 location_id=reserved_quant.location_id.id,
                 lot_id=reserved_quant.lot_id.id or False,
-                package_id=reserved_quant.package_id.id or False,
+                package_id=package.id or False,
                 owner_id =reserved_quant.owner_id.id or False,
             )
+        # apply putaway
+        location_dest_id = self.location_dest_id._get_putaway_strategy(self.product_id, quantity=quantity or 0, package=package, packaging=self.product_packaging_id).id
+        vals = dict(vals, location_dest_id=location_dest_id)
```

Comment:
vals['location_dest_id'] = location_dest_id

recreate a dict each time is not optimal